backend/de.metas.document.archive/de.metas.document.archive.base/src/main/java/de/metas/document/archive/mailrecipient/process/C_Doc_Outbound_Log_SendPDFMails.java

```diff
@@ -40,4 +63,59 @@ public ProcessPreconditionsResolution checkPreconditionsApplicable(@NonNull fina
 
 		return ProcessPreconditionsResolution.accept();
 	}
+
+	@Param(parameterName = PARA_OnlyNotSentMails, mandatory=true)
+	private boolean p_OnlyNotSentMails = false;
+
+	@Override
+	protected final void prepare()
+	{
+		final IQueryFilter<I_C_Doc_Outbound_Log> filter = getFilter();
+		final PInstanceId pinstanceId = getPinstanceId();
+
+		//
+		// Create selection for PInstance and make sure we're enqueuing something
+		final int selectionCount = queryBL.createQueryBuilder(I_C_Doc_Outbound_Log.class, this)
+				.addOnlyActiveRecordsFilter()
+				.filter(filter)
+				.create()
+				.createSelection(pinstanceId);
+
+		if (selectionCount == 0)
+		{
+			throw new AdempiereException(MSG_No_DocOutboundLog_Selection);
+		}
+	}
+
+	protected IQueryFilter<I_C_Doc_Outbound_Log> getFilter()
+	{
+		final ProcessInfo pi = getProcessInfo();
+		final IQueryFilter<I_C_Doc_Outbound_Log> selectedRecordsFilter = pi.getQueryFilterOrElse(ConstantQueryFilter.of(false));
+
+		return queryBL
+				.createCompositeQueryFilter(I_C_Doc_Outbound_Log.class)
+				.addNotNull(I_C_Doc_Outbound_Log.COLUMNNAME_CurrentEMailAddress)
+				.addFilter(selectedRecordsFilter);
+	}
+
+	@Override
+	protected final String doIt() throws Exception
+	{
+		final PInstanceId pinstanceId = getPinstanceId();
+		final MutableInt counter = docOutboundService.sendMail( retrieveSelectedDocOutboundLogs(pinstanceId), pinstanceId, false, p_OnlyNotSentMails);
+
+		return msgBL.getMsg(Async_Constants.MSG_WORKPACKAGES_CREATED, ImmutableList.of(counter.getValue()));
+	}
+
+	private Stream<I_C_Doc_Outbound_Log> retrieveSelectedDocOutboundLogs(final PInstanceId pinstanceId)
+	{
+		final Iterator<I_C_Doc_Outbound_Log> iterator = queryBL
```

Comment:
no queryBL outside DAO

Reply:
didn't get to refactor this part of the original code yet